### PR TITLE
use preprocessing to handle the first time apply with inventorypolicy

### PR DESCRIPTION
**Pull request by Liujingfang1**
With `inventory-policy` flags added, users have two options: `strict` and `adopt`. `strict` means an object can be updated/pruned only when the `owning-inventory` annotation matches with the current inventory object. `adopt` means that an object can be updated/pruned when the `owning-inventory` annotation matches or this annotation is not present. The default option is `strict`. 

When package was applied by previous version of `kpt`, which doesn't have the `inventory-policy` flag. We need to handle the first time apply when user switches to the new version of `kpt`. In the first time apply, the inventory-policy should be `adopt` so that the resources will be added the `owning-inventory` annotation. 

In this change, the first time apply is detected by checking the label "apps.kubernetes.io./managed-by". If this label is missing, we treat it as the first time apply and add this label to the inventory object.

This PR depends on https://github.com/kubernetes-sigs/cli-utils/pull/316

### CODE REVIEW

File: commands/applycmd.go [hunk 1]
```diff
@@ -8,23 +8,30 @@ import (
...
 	"sigs.k8s.io/cli-utils/cmd/apply"
+	"sigs.k8s.io/cli-utils/pkg/common"
+	"sigs.k8s.io/cli-utils/pkg/inventory"
 	"sigs.k8s.io/cli-utils/pkg/manifestreader"
 	"sigs.k8s.io/cli-utils/pkg/provider"
 )
 
+const (
+	inventoryPolicyFlag = "inventory-policy"
+	strictPolicy        = "strict"
```
Comment (seans3) [inline commands/applycmd.go:25]: Does these consts belong here? Or in a more central location (e.g. `pkg/inventory` or `pkg/common`)

**Review by seans3 — COMMENTED** [review]
Small question about consts

File: commands/applycmd.go (hunk 1 above)
Comment (Liujingfang1) [reply commands/applycmd.go]: Agreed that they should be from some common locations. Moved them as constants in cli-utils.

**Comment by Liujingfang1** [comment]
@seans3 Addressed the comments. It's ready for another review.

**Comment by mortent** [comment]
I think this looks good. Would the use of cobra command wrappers be simplified if we just implement the commands in kpt instead of cli-utils?
And is there a way we can add tests for some of this?

**Comment by Liujingfang1** [comment]
> I think this looks good. Would the use of cobra command wrappers be simplified if we just implement the commands in kpt instead of cli-utils?

Yes, it would simplify the implementation a lot. We don't need to inject the preprocessing from the wrapper to the actual cli.

> And is there a way we can add tests for some of this?
I'm also thinking about this. The existing e2e test, specifically the one for testing `kpt live` of different kpt versions, covers the first-time apply case. Besides that, we can add a unit test for the `PreProcess` function.

**Comment by Liujingfang1** [comment]
@mortent I added unit tests for the pre processing function. PTAL

**Comment by seans3** [comment]
looks good to me. I'll let Morten also sign-off.

**Review by mortent — APPROVED** [review]

**Pull request merged**